### PR TITLE
Added database CI tests

### PR DESCRIPTION
**Pull request by Jasperav**
This PR ensures the CI spins up a ScyllaDB instance in Docker and executes database tests

- [X] I have split my patch into logically separate commits.
- [X] All commit messages clearly explain what they change and why.
- [X] I added relevant tests for new features and bug fixes. -> Not needed since the CI now executes ignored tests
- [X] All commits compile, pass static checks and pass test.
- [X] PR description sums up the changes and reasons why they should be introduced.
- [X] I added appropriate `Fixes:` annotations to PR description.

### CODE REVIEW

File: scylla/src/frame/response/result.rs [hunk 1]
```diff
@@ -479,7 +479,7 @@ fn deser_rows(buf: &mut &[u8]) -> StdResult<Rows, ParseError> {
     })
 }
 
-#[allow(clippy::unnecessary_wraps)]
+#[allow(clippy::unnecessary_unwrap)]
```
Comment (Jasperav) [inline scylla/src/frame/response/result.rs:482]: clippy fails on 'unnecessary_wraps'

**Comment by psarna** [comment]
I think it makes more sense to keep all ignored tests as is, and simply ensure that CI runs ignored tests as well. Otherwise all developers would have to have a running scylla instance in order to run `cargo test` locally, and it's not always the case. @piodul?

**Comment by psarna** [comment]
It's true that we kind of misuse the purpose of `--ignored`, but I don't have any better idea for a flag that only runs the extended test suite if explicitly turned on. Does anyone have a better idea?

**Comment by Jasperav** [comment]
@psarna maybe we can do this with features. 

But I don't understand the argument that the tests should be kept separate, because developers don't always have a scylladb instance running. Is that really the case? It is pretty easy to setup Scylla through Docker.

If it isn't easy for some developers to setup Scylla and the ignore method is kept, than tests are not ran for them. Is that any different than failing test? In both scenario's, the developer (which doesn't have a ScyllaDb running) is unable to verify the output of the database test. The only difference is that in one case it is just ignored and in the other case, the test is red.

I think we should assume that developers always have a Scylla instance running (but maybe you know this better than me). It is easier to not distinguish database tests from regular tests, both for CI and local testing.

**Comment by psarna** [comment]
Alright, it makes sense. I actually don't usually have a running Scylla instance (or rather, I do, but for different purposes), but it's not a big issue - the tests which require a database will simply fail. Later we can (and should) make some of the tests opt-out if somebody wants to consciously run them without a running Scylla instance, but that's out of scope for this pull request. Looks good.

**Review by psarna — APPROVED** [review]

**Pull request merged**